service-model-generator/modules/service-model-generator-ls-extension/src/main/java/io/ballerina/servicemodelgenerator/extension/ServiceModelGeneratorService.java

```diff
@@ -0,0 +1,816 @@
+                    OpenApiServiceGenerator oasSvcGenerator = new OpenApiServiceGenerator(contractPath,
+                            project.sourceRoot(), workspaceManager);
+                    return new CommonSourceResponse(oasSvcGenerator.generateService(serviceContractTypeName,
+                            service.getListener().getValue()));
+                }
+
+                String serviceDeclaration = getServiceDeclarationNode(service);
+                TextEdit serviceEdit = new TextEdit(Utils.toRange(lineRange.endLine()),
+                        System.lineSeparator() + serviceDeclaration);
+                if (!importExists(node, service.getOrgName(), service.getModuleName())) {
+                    String importText = String.format("%simport %s/%s;%s", System.lineSeparator(), service.getOrgName(),
+                            service.getModuleName(), System.lineSeparator());
+                    TextEdit importEdit = new TextEdit(Utils.toRange(lineRange.startLine()), importText);
+                    edits.add(importEdit);
+                }
+                edits.add(serviceEdit);
+                return new CommonSourceResponse(Map.of(request.filePath(), edits));
+            } catch (Throwable e) {
+                return new CommonSourceResponse(e);
+            }
+        });
+    }
+
+    @JsonRequest
+    public CompletableFuture<TriggerListResponse> getTriggerModels(TriggerListRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            List<TriggerBasicInfo> triggerBasicInfoList = triggerProperties.values().stream()
+                    .filter(triggerProperty -> filterTriggers(triggerProperty, request))
+                    .map(trigger -> getTriggerBasicInfoByName(trigger.name()))
+                    .flatMap(Optional::stream)
+                    .toList();
+            return new TriggerListResponse(triggerBasicInfoList);
+        });
+    }
+
+    @JsonRequest
+    public CompletableFuture<ResourceModelResponse> getHttpResourceModel() {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                return getResourceFunctionModel()
+                        .map(ResourceModelResponse::new)
+                        .orElseGet(ResourceModelResponse::new);
+            } catch (Throwable e) {
+                return new ResourceModelResponse(e);
+            }
+        });
+    }
+
+    @JsonRequest
+    public CompletableFuture<CommonSourceResponse> addResource(FunctionSourceRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                Path filePath = Path.of(request.filePath());
+                this.workspaceManager.loadProject(filePath);
+                Optional<Document> document = this.workspaceManager.document(filePath);
+                if (document.isEmpty()) {
+                    return new CommonSourceResponse();
+                }
```

Comment:
The API expects to add the resource to the service and if the document is missing, this indicates a program anomaly, and the error should be sent to the front end for debugging. It's best to throw an exception in this case.